### PR TITLE
fix: work with cycle defendant properties

### PR DESCRIPTION
**Pull request by Tenischev**
**Description**

- use `json-decycle` to remove schema cycles for `JSON.stringify`
- during testing found issue with parser, reported here https://github.com/asyncapi/parser-js/issues/293#issuecomment-968347295
- tested with
```
components:
  messages:
    lightMeasured:
      summary: Inform about environmental lighting conditions for a particular streetlight.
      payload:
        $ref: "#/components/schemas/lightMeasuredPayload"
  schemas:
    lightMeasuredPayload:
      type: object
      properties:
        lumens:
          type: integer
          minimum: 0
          description: Light intensity measured in lumens.
        childs:
          type: array
          items:
            $ref: "#/components/schemas/lightMeasuredPayload"
```

**Related issue(s)**
Fixes: #150

### CODE REVIEW

File: filters/all.js [hunk 1]
```diff
@@ -98,7 +99,7 @@ function isDefined(obj) {
 filter.isDefined = isDefined;
 
 function isProtocol(api, protocol){
```
Comment (derberg) [inline filters/all.js:101]: can you remind me what is this filter responsible for?

Comment (Tenischev) [reply filters/all.js:101]: @derberg check if arg `protocol` is mentioned in async-api file as [expected protocol](https://www.asyncapi.com/docs/specifications/v2.0.0#definitionsProtocol). Depend on which protocol user defined in async-api different configurations and dependencies will be generated.

**Comment by derberg** [comment]
@Tenischev as @magicmatatjahu recently released new parser-js with function dedicated to stringify securely (handles circular deps like a pro) AsyncAPI document, and it is already in the generator:
- you can remove `decycle`
- in `package.json` update that template will work only with generator `"generator": ">=1.8.27 <2.0.0",`
- this is how you access the stringify function now `asyncapi.constructor.stringify(asyncapi)`

**Comment by derberg** [comment]
@Tenischev btw mate, I see you are still using the template because you are still improving it, don't you want to join as a maintainer? so except for pushing your stuff, you would have to also support us on review or other PRs and triaging existing issues

**Comment by Tenischev** [comment]
@derberg thanks for the advice, I redid it

And regarding your suggestions, yes it would be interesting for me.

btw, Merry Christmas! 🎅

**Comment by derberg** [comment]
@Tenischev 

Merry Christmassssss!

Please also remove `json-decycle` from package.json, and also changes in `package-lock.json` should not be reflected in this PR.

> And regarding your suggestions, yes it would be interesting for me.

Perfect, do you know anyone who would like to join you, or would it have to be me until we find someone?

**Comment by derberg** [comment]
@Tenischev yo man, back at work already? 😄

**Comment by Tenischev** [comment]
Not yet :), tomorrow last day of vacation
I will fix it within next days

**Comment by derberg** [comment]
oh lucky you!!! enjoy!

**Comment by Tenischev** [comment]
@derberg `package` and `package-lock` are fixed.

> Perfect, do you know anyone who would like to join you, or would it have to be me until we find someone?

Unfortunately, no, so it's going to be you 😄

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
@Tenischev 😭 😆 please open up a PR to add codeowners and add us there. Also specify if you can help us on org level and want to join TSC 😉

**Pull request merged**